### PR TITLE
Preprocess: handle Ranges placeholders and avoid unnecessary regex use

### PR DESCRIPTION
**Pull request by PeterFeicht**
This handles the new Ranges placeholder links, which are replaced by JS on the site but remain absolute when downloaded (because they result in a 404 status) and replaces them by proper relative links during HTML preprocessing.

There's also some small changes to regular expression use: use raw string literals for regex strings, and avoid unnecessary regex use which can be replaced by simple string operations.

### CODE REVIEW

File: commands/preprocess.py [hunk 1]
```diff
@@ -173,6 +173,28 @@ def transform_loader_link(target, file, root):
     abstarget = os.path.join(root, "common/" + convert_loader_name(target))
     return os.path.relpath(abstarget, os.path.dirname(file))
 
+def is_ranges_placeholder(target):
```
Comment (p12tic) [inline commands/preprocess.py]: Can we have a unit test for this function? `tests/test_preprocessor.py` might be a good place.

File: commands/preprocess.py [hunk 2]
```diff
@@ -173,6 +173,28 @@ def transform_loader_link(target, file, root):
...
+    if re.match(r'https?://[a-z]+\.cppreference\.com/w/cpp/ranges(-[a-z]+)?-placeholder/.+', target):
+        return True
+    return False
+
+def transform_ranges_placeholder(target, file, root):
```
Comment (p12tic) [inline commands/preprocess.py]: Can we have a unit test for this function? I think in this particular case it's not a problem that we do path operations, as the result is independent on the filesystem state.

**Comment by PeterFeicht** [comment]
Hm, I can add some tests alright.
As for the commit messages, I'd have to rebase and then force push the branch, right? Do you know how I'd handle that when I already merged the branch into my master?

**Comment by p12tic** [comment]
Have you pushed the merge to your public repository? If not, you could rewind your master to just before the merge and then redo the merge with fixed commits. And even if you did, it's not a problem to force push if it's unlikely that someone has pulled it in the meantime.

**Comment by PeterFeicht** [comment]
You're probably right, it's unlikely anyone has pulled let alone based work on the new master. I'll make the changes shortly.

**Comment by PeterFeicht** [comment]
Done, I also added a note to test and preprocess code on where the placeholder replacement JS can be found for the main site.

**Pull request merged**